### DISCUSSION TITLE
Get mesh region settings usage

### DISCUSSION THREAD

**Discussion by pruthvi1991** [ansys/pyaedt#4915] [Q&A] [answered]
Hi,

I have an icepak project where I have assigned mesh settings using pyaedt. Later I repoened this project using pyaedt and tried to read the mesh settings. I see different values between the GUI and what the API outputs. See below screenshot showing the GUI read settings corresponding to the global region

![image](https://github.com/user-attachments/assets/a501a02a-ff37-4772-a908-7d9274a58e56)

Here is what I read from the API:

```
all_mesh_regions = ipk.mesh.meshregions
vars(all_mesh_regions[0])
```

```
{'global_region': <pyaedt.modules.MeshIcepak.Region at 0x1a75b5c1fd0>,
 'manual_settings': False,
 'settings': {'Enable2DCutCell': False, 'CurvatureSizeFunction': True, 'ProximitySizeFunction': True, 'Enforce2dot5DCutCell': False, 'EnforceCutCellMeshing': False, 'StairStepMeshing': False, 'EnableTransition': False, 'MeshRegionResolution': 3, 'OptimizePCBMesh': True},
 '_name': 'Settings',
 '_model_units': 'mm',
 '_app': <pyaedt.icepak.Icepak at 0x1a755eab810>}
```

Am I not using the object methods/attributes correctly? Any help appreciated.

**Comment by lorenzovecchietti** [comment]
Hi, did you have a chance to review [this](https://aedt.docs.pyansys.com/version/stable/User_guide/mesh.html#mesh-in-icepak)?

When you are done with the mesh settings modification, you can run the update command

**Comment by pruthvi1991** [comment]
@lorenzovecchietti I did go over it a while ago. My issue is not with assigning mesh settings, it's about reading the assigned settings. I have created several mesh regions using a code block very similar to this.

```
global_mesh_settings = {'MaxElementSizeZ': '50um', 
                       'MaxElementSizeX': '50um', 
                       'MaxElementSizeY': '50um', 
                       'MinGapX': '1um',
                       'MinGapY': '1um', 
                       'MinGapZ': '1um',
                       'MaxLevels': '1',
                       'NoOGrids': True,
                       'EnforeMLMType': 'None', 
                       }

glob_msh = ipk.mesh.global_mesh_region

glob_msh.UserSpecifiedSettings = True
for key,value in global_mesh_settings.items():
        glob_msh.UserSpecifiedSettings = True
        if hasattr(glob_msh, key):
            setattr(glob_msh,key,value)

glob_msh.update()

```

I analyzed that project and saved it. Later I opened the same project using the GUI and tried to output the mesh settings within pyaedt and I don't see the values that the API outputs match the values I see in the GUI.

```
ipk = Icepak()
glob_msh = ipk.mesh.global_mesh_region
glob_msh.settings

```
{'MeshRegionResolution': 3, 'CurvatureSizeFunction': True, 'OptimizePCBMesh': True, 'EnforceCutCellMeshing': False, 'ProximitySizeFunction': True, 'EnableTransition': False, 'StairStepMeshing': False, 'Enforce2dot5DCutCell': False, 'Enable2DCutCell': False}

**Reply by lorenzovecchietti** [reply]
Oh, sorry I misread your first question. Yes, I'm able to reproduce the issue, and I'm pushing a fix soon.

**Reply by lorenzovecchietti** [reply, marked as answer]
PR #4923

**Comment by pruthvi1991** [comment]
Works as intended after the PR #4923 from @lorenzovecchietti

Output from api matches GUI value

`{'NoOGrids': True, 'Enable2DCutCell': False, 'StairStepMeshing': False, 'EnableMLM': False, 'UniformMeshParametersType': 'None', 'EnforeMLMType': 'None', 'EnforceCutCellMeshing': False, '2DMLMType': '2DMLM_None', 'Enforce2dot5DCutCell': False, 'MaxElementSizeX': '25um', 'MaxLevels': '1', 'EnableTransition': False, 'MinGapX': '10um', 'OptimizePCBMesh': True, 'ProximitySizeFunction': True, 'MinGapZ': '10um', 'MinElementsOnEdge': '1', 'MinElementsInGap': '1', 'BufferLayers': '0', 'CurvatureSizeFunction': True, 'MaxElementSizeY': '25um', 'MaxSizeRatio': '2', 'MinGapY': '10um', 'MaxElementSizeZ': '25um'}`
